elf_reader.go

```diff
@@ -592,7 +592,7 @@ func (ec *elfCode) loadBTFMaps(maps map[string]*MapSpec) error {
 				return fmt.Errorf("expected struct, got %s", v.Type)
 			}
 
-			mapSpec, err := mapSpecFromBTF(sec, &vs, mapStruct, ec.btf, name, false)
+			mapSpec, err := mapSpecFromBTF(sec, &ds.Vars[idx], mapStruct, ec.btf, name, false)
```

Comment:
The memory aliasing warning is pretty convenient while coding IMO, but that's pretty much it. A bit too many exclusions for my liking.

Reply:
> The memory aliasing warning is pretty convenient while coding IMO

All contributors may not run linters with checks like this so having CI error for them might be useful.